exercises/simple-linked-list/tests/simple-linked-list.rs

```diff
@@ -0,0 +1,89 @@
+extern crate simple_linked_list;
+
+use simple_linked_list::SimpleLinkedList;
+
+#[test]
+fn test_new_list_is_empty() {
+    let list: SimpleLinkedList<u32> = SimpleLinkedList::new();
+    assert_eq!(list.len(), 0, "list's length must be 0");
+}
+
+#[test]
+#[ignore]
+fn test_push_increments_length() {
+    let mut list: SimpleLinkedList<u32> = SimpleLinkedList::new();
+    list.push(1);
+    assert_eq!(list.len(), 1, "list's length must be 1");
+    list.push(2);
+    assert_eq!(list.len(), 2, "list's length must be 2");
+}
+
+#[test]
+#[ignore]
+fn test_pop_decrements_length() {
+    let mut list: SimpleLinkedList<u32> = SimpleLinkedList::new();
+    list.push(1);
+    list.push(2);
+    list.pop();
+    assert_eq!(list.len(), 1, "list's length must be 1");
+    list.pop();
+    assert_eq!(list.len(), 0, "list's length must be 0");
+}
+
+#[test]
+#[ignore]
+fn test_pop_returns_last_added_element() {
+    let mut list: SimpleLinkedList<u32> = SimpleLinkedList::new();
+    list.push(1);
+    list.push(2);
+    assert_eq!(list.pop(), Some(2), "Element must be 2");
+    assert_eq!(list.pop(), Some(1), "Element must be 1");
+    assert_eq!(list.pop(), None, "No element should be contained in list");
+}
+
+#[test]
+#[ignore]
+fn test_peek_returns_head_element() {
+    let mut list: SimpleLinkedList<u32> = SimpleLinkedList::new();
+    assert_eq!(list.peek(), None, "No element should be contained in list");
+    list.push(2);
+    assert_eq!(list.peek(), Some(&2), "Element must be 2");
+    assert_eq!(list.peek(), Some(&2), "Element must be still 2");
+}
+
+#[test]
+#[ignore]
+fn test_from_slice() {
+    let array = ["1", "2", "3", "4"];
+    let mut list = SimpleLinkedList::from(array.as_ref());
+    assert_eq!(list.pop(), Some("1"));
+    assert_eq!(list.pop(), Some("2"));
+    assert_eq!(list.pop(), Some("3"));
+    assert_eq!(list.pop(), Some("4"));
+}
```

Comment:
The tests as written reverse the order of an input slice, and again when emitting a vector. While that's _consistent_, in that `vec![1, 2, 3] == SimpleLinkedList::from([1, 2, 3].as_ref()).into()`, I'd expect that to cause some confusion for students. If you build a vec and a list in the same way, they should be the same:

```rust
let mut v = Vec::new();
let mut s = SimpleLinkedList::new();
for i in vec![1, 2, 3] {
    v.push(i);
    s.push(i);
}
assert_eq!(v, s.into());
```

Please change the semantics of the traits converting from slices and to vectors so that they preserve the order of elements, and update the tests accordingly.